proxysql/datadog_checks/proxysql/proxysql.py

```diff
@@ -0,0 +1,297 @@
+
+        if 'query_rules_metrics' in additional_metrics:
+            query_rules_stats = self._get_query_rules_stats(conn)
+            self._send_extra_tag_metrics(query_rules_stats, STATS_MYSQL_QUERY_RULES, tags)
+
+    def _send_simple_tag_metrics(self, stats, metrics_definition, tags):
+        for proxysql_metric_name, metric_details in metrics_definition.items():
+            metric_name, metric_type = metric_details
+            metric_tags = list(tags)
+            self._send_metric(metric_name, metric_type, float(stats.get(proxysql_metric_name)), metric_tags)
+
+    def _send_extra_tag_metrics(self, stats, metrics_definition, tags):
+        for proxysql_metric_name, metric_details in metrics_definition.items():
+            metric_name, metric_type = metric_details
+
+            for metric in stats.get(proxysql_metric_name, []):
+                metric_tags = list(tags)
+                tag, value = metric
+                metric_tags.append(tag)
+                self._send_metric(metric_name, metric_type, float(value), metric_tags)
+
+    def _send_metric(self, metric_name, metric_type, metric_value, metric_tags):
+        if metric_type == RATE:
+            self.rate(metric_name, metric_value, tags=metric_tags)
+        elif metric_type == GAUGE:
+            self.gauge(metric_name, metric_value, tags=metric_tags)
+        elif metric_type == COUNT:
+            self.count(metric_name, metric_value, tags=metric_tags)
+        elif metric_type == MONOTONIC:
+            self.monotonic_count(metric_name, metric_value, tags=metric_tags)
+
+    def _fetch_stats(self, conn, query, stats_name):
+        try:
+            with closing(conn.cursor()) as cursor:
+                cursor.execute(query)
+
+                if cursor.rowcount < 1:
+                    self.warning("Failed to fetch records from %s.", stats_name)
+                    return []
+
+                return cursor.fetchall()
+        except (pymysql.err.InternalError, pymysql.err.OperationalError) as e:
+            self.warning("ProxySQL %s unavailable at this time: %s", stats_name, str(e))
+            return []
+
+    def _get_global_stats(self, conn):
+        """Fetch the global ProxySQL stats."""
+        query = "SELECT * FROM stats.stats_mysql_global"
+        return {row["Variable_Name"]: row["Variable_Value"] for row in self._fetch_stats(conn, query, 'global_stats')}
+
+    def _get_command_counters(self, conn):
+        """Fetch ProxySQL command counters stats"""
+        query = "SELECT * FROM stats.stats_mysql_commands_counters"
+        metrics = defaultdict(list)
+
+        for row in self._fetch_stats(conn, query, 'command_counters_stats'):
+            metrics["Total_Time_ms"].append(
+                ("proxysql_command:%s" % row["Command"], str(float(row["Total_Time_us"]) / 1000))
+            )
+            metrics["Total_cnt"].append(("proxysql_command:%s" % row["Command"], row["Total_cnt"]))
```

Comment:
```suggestion
 for row in self._fetch_stats(conn, query, 'command_counters_stats'):
            extra_tag = "proxysql_command:%s" % row["Command"]
            metrics["Total_Time_ms"].append(
                (extra_tag, str(float(row["Total_Time_us"]) / 1000))
            )
            metrics["Total_cnt"].append((extra_tag, row["Total_cnt"]))
```
I see `"proxysql_command:%s" % row["Command"]` tag used frequently, can we assign it to a variable? Same for the tags we use frequently in `_get_connection_pool_stats()` and `_get_user_stats()`